File: filter-pattern/python/src/CriteriaSingle.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""
from src.Criteria import Criteria


# 定义子类覆写父类抽象方法
class CriteriaSingle(Criteria):

    def filter(self, persons):
        single_persons = []
        for person in persons:
            if person.get_status().upper() == 'SINGLE':
                single_persons.append(person)

        return single_persons

File: filter-pattern/go/src/CriteriaSingle.go
package src

import (
  "strings"
)

// 根据标准接口实现的过滤
type CriteriaSingle struct {
}

// 根据是否单身进行过滤
func (c *CriteriaSingle) Filter(persons []*Person) []*Person {
  var singlePersons []*Person
  for _, person := range persons {
    if strings.ToUpper(person.GetStatus()) == "SINGLE" {
      singlePersons = append(singlePersons, person)
    }
  }
  return singlePersons
}
